public/js/highlight-sequence.js

```diff
@@ -50,30 +46,137 @@ AFRAME.registerComponent("highlight-sequence", {
-      console.log("highlighting " + sequenceName);
+    const cleanupOutlines = () => {
+      const outlineElems = document.querySelectorAll("[isoutline='true']");
+      outlineElems.forEach((elem) => {
+        elem.parentElement.removeChild(elem);
+      });
+    };
+    const cleanupHighlights = (sequenceName = 'all_pathways') => {
       const nodes = View.sequences.nodes[sequenceName];
       const edges = View.pathways[sequenceName];
       if (nodes != undefined) {
         for (let i = 0; i < nodes.length; i++) {
           const metabolite = nodes[i];
           console.log("HIGHLIGHT" + metabolite);
-          colorNode(metabolite);
+          colorNode(metabolite, INITIAL_NODE_COLOR);
           const outputMetabolite = nodes[i + 1];
           if (outputMetabolite !== null) {
-            colorEdge(metabolite, outputMetabolite);
+            colorEdge(metabolite, outputMetabolite, INITIAL_EDGE_COLOR);
           }
         }
       }
       if (edges != undefined) {
         edges.forEach((edge) => {
-          colorEdge(edge.input, edge.output);
-          [edge.input, edge.output].forEach(colorNode);
+          colorEdge(edge.input, edge.output, INITIAL_EDGE_COLOR);
+          [edge.input, edge.output].forEach((node) => colorNode(node, INITIAL_NODE_COLOR));
         });
       }
+    }
+    const updateButtonColor = (clickState, sequenceName) => {
+      const button = document.getElementById(sequenceName + '_button');
+      // button attibutes seem to only take hex codes. 'red', 'purple', etc don't work.
+      switch (clickState) {
+        case 0: {
+          // default colors: https://github.com/rdub80/aframe-gui#a-gui-button-component
+          button.setAttribute("background-color", '#22252a');
+          // TODO - these don't work..
+          // button.setAttribute("border-color", '#d3d3d4');
+          // button.setAttribute("font-color", '#d3d3d4');
+
+          return;
+        }
+        case 1: {
+          button.setAttribute("background-color", '#FF0000');
+          // TODO - these don't work..
+          // button.setAttribute("font-color", '#ffffff');
+          // button.setAttribute("border-color", '#d3d3d4');
+
+          return;
+        }
+        case 2: {
+          button.setAttribute("background-color", '#00FFFF');
+
+          return;
+        }
+        default: {
+          return;
+        }
+      }
+    }
+    this.el.addEventListener('click', () => {
+      const component = this.el.getAttribute("highlight-sequence");
+      const sequenceName = component.sequence;
+      if (component.clickState === 2) {
+        cleanupOutlines();
+      }
+      const clickState = (component.clickState + 1) % 3;
+      updateButtonColor(clickState, sequenceName);
+      // increment clickState attribute
+      this.el.setAttribute('highlight-sequence', 'clickState', clickState);
+      switch (clickState) {
+        case 0:
+          cleanupHighlights(sequenceName);
```

Comment:
Calling this function here makes highlights for edges in other sequences disappear too. Should we call cleanupOutlines here instead?

Reply:
![image](https://user-images.githubusercontent.com/13208224/96904404-ec084600-144b-11eb-847f-801c47938f19.png)
![image](https://user-images.githubusercontent.com/13208224/96904453-ffb3ac80-144b-11eb-9532-cf247f6f03f0.png)
![image](https://user-images.githubusercontent.com/13208224/96904488-0c380500-144c-11eb-8aba-11248c2c341d.png)
![image](https://user-images.githubusercontent.com/13208224/96904557-1fe36b80-144c-11eb-92f7-e480a63b94f5.png)

Reply:
I see now that calling just cleanupOutlines won't get rid of the highlight on the bottom 2 edges. I think we may need a data structure to keep track of highlighted edges or make edges keep track of their state. I will draw up a design to present at our meeting this afternoon.

Reply:
yep that makes sense, since the sequences can overlap between pathways